api/types/helmchartargs.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// HelmChartArgs contains the metadata of how to generate a secret.
+type HelmChartArgs struct {
+	ChartName        string `json:"chartName,omitempty" yaml:"chartName,omitempty"`
+	ChartVersion     string `json:"chartVersion,omitempty" yaml:"chartVersion,omitempty"`
+	ChartRepoURL     string `json:"chartRepoUrl,omitempty" yaml:"chartRepoUrl,omitempty"`
```

Comment:
`ChartHome`, and most (all?) other fields from the bash example, and keep the original names to ease migration.
User should only have to change `apiVersion` in the config file.

i think we need to retain the ability to use local charts (that's how they are developed)

Most of the fields in the bash script were added for particular use cases, by various users.
e.g.  `HelmBin` was added to avoid the need to adjust `PATH` in certain contexts.

We can wait to add this in another pr, since we're also going to retain the old plugin for a while.